src/main/java/org/variantsync/diffdetective/variationtree/VariationTreeNode.java

```diff
@@ -0,0 +1,364 @@
+ * this implementation optimizes for variation trees obtained from source code annotated with the C
+ * preprocessor syntax. This mostly means that there is the additional {@link NodeType#ELIF} node
+ * type.
+ *
+ * <p>This class contains references to all of its children and its parent so all connected nodes of
+ * a variation tree can be reached through each node of the variation tree. Nevertheless, most of
+ * the time only node itself or its subtree (the reflexive hull of {@link getChildren}) is meant
+ * when when referencing a {@code VariationTreeNode}. Use {@link VariationTree} to unambiguously
+ * refer to a whole variation tree.
+ *
+ * <p>If possible, algorithms should be using {@link VariationNode} instead of this concrete
+ * implementation. This allows the usage of a projection of a {@link DiffNode} instead of a concrete
+ * variation node.
+ *
+ * <p>To compare two variation trees use {@link DiffTree} which uses the aforementioned
+ * {@link DiffNode}s.
+ *
+ * @see VariationTree
+ * @see DiffTree
+ * @see DiffNode
+ * @author Benjamin Moosherr
+ */
+public class VariationTreeNode extends VariationNode<VariationTreeNode> {
+    private static final short ID_OFFSET = 3;
+
+    /**
+     * The node type of this node, which determines the type of the represented
+     * element in the diff (e.g., mapping or artifact).
+     */
+    public final NodeType nodeType;
+
+    /**
+     * The start line number of the {@link label} of this node in the corresponding source code.
+     */
+    private int from = DiffLineNumber.InvalidLineNumber;
+    /**
+     * The end line number of the {@link label} of this node in the corresponding source code.
+     * The line number is exclusive (i.e., it points 1 behind the last included line).
+     */
+    private int to = DiffLineNumber.InvalidLineNumber;
+
+    /**
+     * A list of lines representing the label of this node.
+     */
+    private List<String> label;
+
+    /**
+     * The direct feature mapping of this node.
+     *
+     * <p>This is {@code null} iff {@link isConditionalAnnotation} is {@code false}.
+     */
+    private Node featureMapping;
+
+    /**
+     * The parent of this node. It's {@code null} iff this node doesn't have a parent.
+     *
+     * <p>Invariant: Iff {@code parent != null} then {@code parent.childOrder.contains(this)}.
+     */
+    private VariationTreeNode parent;
+
+    /**
+     * The list for maintaining the order of all children of this node.
+     *
+     * @see parent
+     */
+    private final List<VariationTreeNode> childOrder;
+
+    /**
+     * Creates a new node of a variation tree.
+     *
+     * The newly created node is not connected to any other nodes.
+     *
+     * The given label is not copied, so modifications to {@code label} reflect in the label of this
+     * node. It's recommended to remove all references to {@code label} after the node is fully
```

Comment:
I do not understand `modifications to label reflect in the label of this node`.

Reply:
What about saying "takes ownership of the list without copying"?

Reply:
Thanks for this formulation. I struggled quite a bit to get decent wording and then just gave up.

Reply:
ok :D